### PR TITLE
refactor(connector): [CyberSource] Enhance currency Mapping with ConnectorCurrencyCommon Trait

### LINKED ISSUE #2222: [FEATURE]: [Cybersource] Currency Unit Conversion
### :memo: Feature Description

- Each currency can be described in terms of base or minor units. 
- For instance, when dealing with USD currency, the Cent is considered as the minor unit, while the Dollar is considered the base unit. 
- In Hyperswitch, the amount value is expected to be always provided in minor units. 
- For example: In USD, If the amount is 1 in base unit (Dollar), then it will be equivalent to 100 in minor units (Cent)
- Some of the connectors integrated require the amount to be converted to a desirable unit before being passed to them. 
- We have functions `to_currency_base_unit` and `to_currency_lower_unit` in place to convert the minor unit amount to its decimal equivalent and vice versa. These conversions are handled explicitly based on the connector.

### :hammer: Possible Implementation

- ConnectorCommon trait have been implemented for the connector. 
- This trait contains `get_currency_unit` method. This method needs to be implemented. 
- It will define what type of conversion needs to be done during `connector_router_data` creation. Concurrently handle the creation of `connector_router_data` to be passed for the connector request body creation.
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/2196

:bookmark: Note: All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by mdrokz**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
This PR implements `get_currency_unit` for cybersource to convert payments from `Base` to `Minor` it also implements router data to handle the conversion.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
Closes #2222 

## How did you test it?

I tested by sending a payment request & verifying the amount is showed in minor units in the cybersource console

image

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by SanchithHegde** [comment]
Hey @mdrokz, could you address the failing CI checks?

**Comment by mdrokz** [comment]
> Hey @mdrokz, could you address the failing CI checks?

Just synced the changes from main can you run the CI checks again ?

**Comment by SanchithHegde** [comment]
@mdrokz The formatting check is failing, could you please address it?

**Comment by mdrokz** [comment]
> @mdrokz The formatting check is failing, could you please address it?

Done

**Comment by mdrokz** [comment]
@SanchithHegde can you review this ?

**Review by srujanchikke — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/cybersource/transformers.rs [hunk 1]
```diff
@@ -109,6 +140,70 @@ fn build_bill_to(
...
+                })
+            }
+            _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
+        }
+    }
+}
+
 impl TryFrom<&types::PaymentsAuthorizeRouterData> for CybersourcePaymentsRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
```
[the remark this answers is not in the document]
Comment (mdrokz) [reply crates/router/src/connector/cybersource/transformers.rs]: > 

Done i removed the old try form since both of them were same.

**Review by srujanchikke — APPROVED** [review]
LGTM

**Review by Sangamesh26 — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @mdrokz!

**Comment by swangi-kumari** [comment]
Hey @mdrokz ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.  
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.  
Thanks!

**Comment by mdrokz** [comment]
> Hey @mdrokz ,
> Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.  
> Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.  
> Thanks!

Hey thanks a lot for this let me fill it up :)

**Pull request merged**